src/Mod/PartDesign/Gui/TaskLinearPatternParameters.cpp

```diff
@@ -184,41 +105,84 @@ void TaskLinearPatternParameters::setupUI()
         this->fillAxisCombo(dirLinks, nullptr);
     }
 
-    //show the parts coordinate system axis for selection
-    PartDesign::Body * body = PartDesign::Body::findBodyOf(getObject());
-    if(body) {
+    // show the parts coordinate system axis for selection
+    PartDesign::Body* body = PartDesign::Body::findBodyOf(getObject());
+    if (body) {
         try {
-            App::Origin *origin = body->getOrigin();
+            App::Origin* origin = body->getOrigin();
             ViewProviderOrigin* vpOrigin;
-            vpOrigin = static_cast<ViewProviderOrigin*>(Gui::Application::Instance->getViewProvider(origin));
+            vpOrigin = static_cast<ViewProviderOrigin*>(
+                Gui::Application::Instance->getViewProvider(origin));
             vpOrigin->setTemporaryVisibility(true, false);
-        } catch (const Base::Exception &ex) {
-            Base::Console().Error ("%s\n", ex.what () );
+        }
+        catch (const Base::Exception& ex) {
+            Base::Console().Error("%s\n", ex.what());
         }
     }
 
     adaptVisibilityToMode();
-    connectSignals();
+
+    updateViewTimer = new QTimer(this);
+    updateViewTimer->setSingleShot(true);
+    updateViewTimer->setInterval(getUpdateViewTimeout());
+    connect(updateViewTimer,
+            &QTimer::timeout,
+            this,
+            &TaskLinearPatternParameters::onUpdateViewTimer);
+
+    connect(ui->comboDirection,
+            qOverload<int>(&QComboBox::activated),
+            this,
+            &TaskLinearPatternParameters::onDirectionChanged);
+    connect(ui->checkReverse,
+            &QCheckBox::toggled,
+            this,
+            &TaskLinearPatternParameters::onCheckReverse);
+    connect(ui->comboMode,
+            qOverload<int>(&QComboBox::activated),
+            this,
+            &TaskLinearPatternParameters::onModeChanged);
+    connect(ui->spinLength,
+            qOverload<double>(&Gui::QuantitySpinBox::valueChanged),
+            this,
+            &TaskLinearPatternParameters::onLength);
+    connect(ui->spinOffset,
+            qOverload<double>(&Gui::QuantitySpinBox::valueChanged),
+            this,
+            &TaskLinearPatternParameters::onOffset);
+    connect(ui->spinOccurrences,
+            &Gui::UIntSpinBox::unsignedChanged,
+            this,
+            &TaskLinearPatternParameters::onOccurrences);
+}
+
+void TaskLinearPatternParameters::retranslateParameterUI(QWidget* widget)
+{
+    ui->retranslateUi(widget);
 }
 
 void TaskLinearPatternParameters::updateUI()
 {
-    if (blockUpdate)
+    if (blockUpdate) {
         return;
+    }
     blockUpdate = true;
 
-    PartDesign::LinearPattern* pcLinearPattern = static_cast<PartDesign::LinearPattern*>(getObject());
-    PartDesign::LinearPatternMode mode = static_cast<PartDesign::LinearPatternMode>(pcLinearPattern->Mode.getValue());
+    PartDesign::LinearPattern* pcLinearPattern =

```

Comment:
```suggestion
    auto pcLinearPattern =
```
will take care of the linter complaint